### PR TITLE
Remove hardcoding

### PR DESCRIPTION
**Pull request by tgurbich**
- Removed species name hardcoding
- Added Archaea support
- Bug fixes and file renaming

### CODE REVIEW

File: bin/identify_kingdom.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+# Copyright 2023 EMBL - European Bioinformatics Institute
```
Comment (mberacochea) [inline bin/identify_kingdom.py:4]:
```suggestion
# Copyright 2023-2024 EMBL - European Bioinformatics Institute
```

File: bin/identify_kingdom.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+
+def main(taxid, outfile):
+    kingdom = "Bacteria"
+    if not taxid.isdigit():
+        sys.exit("Invalid format of taxid {}".format(taxid))
+    try:
+        url = "https://www.ebi.ac.uk/ena/taxonomy/rest/tax-id/{}".format(taxid)
+        r = run_request(url)
+        res = json.loads(r.text)
```
Comment (mberacochea) [inline bin/identify_kingdom.py:36]: The response from this endpoint is already json, so you can just use `r.json()` to get the json thingy

Comment (tgurbich) [reply bin/identify_kingdom.py]: Thanks! Fixed it.

File: conf/ebi_codon.config [hunk 3]
```diff
@@ -14,15 +14,15 @@ params {
 
     antismash_db = "/hps/nobackup/rdf/metagenomics/service-team/ref-dbs/antismash/7.1.x"
 
-    dbcan_db = "/hps/nobackup/rdf/metagenomics/service-team/ref-dbs/dbcan/4.0"
+    dbcan_db = "/hps/nobackup/rdf/metagenomics/service-team/ref-dbs/dbcan/4.1.3-V12"
```
Comment (mberacochea) [inline conf/ebi_codon.config:17]: We need to update the auto-download bit and uplod this version to the FTP

File: modules/local/lookup_kingdom.nf [hunk 4]
```diff
@@ -0,0 +1,26 @@
+process LOOKUP_KINGDOM {
+
+    tag "${meta.prefix}"
+
+    label 'process_light'
+
+    container 'quay.io/microbiome-informatics/genomes-pipeline.python3base:v1.1'
```
Comment (mberacochea) [inline modules/local/lookup_kingdom.nf:7]: I'll create a different base container, not linked to the genomes-pipeline (but we can swap that later)

File: modules/local/lookup_kingdom.nf [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+    input:
+    tuple val(meta), path(fasta)
+
+    output:
+    tuple val(meta), file("${meta.prefix}_kingdom.txt"), emit: detected_kingdom
```
Comment (mberacochea) [inline modules/local/lookup_kingdom.nf:13]:
```suggestion
    tuple val(meta), path("${meta.prefix}_kingdom.txt"), emit: detected_kingdom
```
`path` is preferred to `file` https://www.nextflow.io/docs/latest/process.html#input-type-file

File: conf/ebi_codon.config (hunk 3 above)
Comment (tgurbich) [reply conf/ebi_codon.config:17]: It is on FTP

File: bin/identify_kingdom.py [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+    except:
+        logging.warning(
+            "Unable to identify lineage for taxid {}. " "Reporting default kingdom instead: Bacteria.".format(taxid)
+        )
+    logging.info("Reporting kingdom {} for taxid {}".format(kingdom, taxid))
+    if outfile:
+        with open(outfile, "w") as file_out:
+            file_out.write(kingdom)
+    else:
+        print(kingdom)
```
[the remark this answers is not in the document]
Comment (tgurbich) [reply bin/identify_kingdom.py:57]: I would prefer this to just print the word to make the script more usable and not have to parse the kingdom out of `INFO:root:Archaea` if the script is used without an output file elsewhere.

File: bin/identify_kingdom.py [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+        if rank == "superkingdom":
+            lineage = res.get("scientificName", "")
+        if lineage.startswith("Archaea"):
+            kingdom = "Archaea"
+        elif lineage.startswith("Vir"):
+            kingdom = "Viruses"
+        elif lineage.startswith("Bac"):
+            pass
+        else:
+            logging.warning("Unknown lineage {}. Reporting default kingdom instead: Bacteria.".format(lineage))
```
[the remark this answers is not in the document]
Comment (tgurbich) [reply bin/identify_kingdom.py]: done

File: bin/identify_kingdom.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (mberacochea) [reply bin/identify_kingdom.py:57]: OK, I agree

File: conf/ebi_codon.config (hunk 3 above)
Comment (mberacochea) [reply conf/ebi_codon.config:17]: Perfect, I'll update this while doing the end-to-end test case

**Review by mberacochea — APPROVED** [review]
image

**Pull request merged**